src/source/Crypto/Dtls_mbedtls.c

```diff
@@ -229,6 +234,32 @@ STATUS dtlsTransmissionTimerCallback(UINT32 timerID, UINT64 currentTime, UINT64
     return retStatus;
 }
 
+#if MBEDTLS_VERSION_NUMBER >= 0x03000000
+VOID dtlsSessionKeyDerivationCallback(PVOID customData, mbedtls_ssl_key_export_type secret_type, const unsigned char* pMasterSecret,
+                                      SIZE_T pMasterSecretLen, const unsigned char clientRandom[MAX_DTLS_RANDOM_BYTES_LEN],
+                                      const unsigned char serverRandom[MAX_DTLS_RANDOM_BYTES_LEN], mbedtls_tls_prf_types tlsProfile)
+{
+    ENTERS();
+
+    /* We're only interested in the TLS 1.2 master secret */
+    if (secret_type != MBEDTLS_SSL_KEY_EXPORT_TLS12_MASTER_SECRET) {
+        DLOGE("Secret type is not matching...");
+    }
+
+    PDtlsSession pDtlsSession = (PDtlsSession) customData;
```

Comment:
As this part is the same as below, wondering if we can just have one of them call the other one or maybe factor it out to a common method.

Reply:
The signature itself is different in both the mbedtls versions. I found it cleaner to maintain both separate.